exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1110 @@
+	"github.com/thrasher-corp/gocryptotrader/exchanges/protocol"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream/buffer"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/trade"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+// GetDefaultConfig returns a default exchange config
+func (d *Deribit) GetDefaultConfig() (*config.Exchange, error) {
+	d.SetDefaults()
+	exchCfg := new(config.Exchange)
+	exchCfg.Name = d.Name
+	exchCfg.HTTPTimeout = exchange.DefaultHTTPTimeout
+	exchCfg.BaseCurrencies = d.BaseCurrencies
+	err := d.SetupDefaults(exchCfg)
+	if err != nil {
+		return nil, err
+	}
+	if d.Features.Supports.RESTCapabilities.AutoPairUpdates {
+		err := d.UpdateTradablePairs(context.Background(), true)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return exchCfg, nil
+}
+
+// SetDefaults sets the basic defaults for Deribit
+func (d *Deribit) SetDefaults() {
+	d.Name = "Deribit"
+	d.Enabled = true
+	d.Verbose = true
+	d.API.CredentialsValidator.RequiresKey = true
+	d.API.CredentialsValidator.RequiresSecret = true
+
+	requestFmt := &currency.PairFormat{Uppercase: true, Delimiter: currency.DashDelimiter}
+	configFmt := &currency.PairFormat{Uppercase: true, Delimiter: currency.DashDelimiter}
+	err := d.SetGlobalPairsManager(requestFmt, configFmt, asset.Futures, asset.Options, asset.OptionCombo, asset.FutureCombo)
+	if err != nil {
+		log.Errorln(log.ExchangeSys, err)
+	}
+
+	// Fill out the capabilities/features that the exchange supports
+	d.Features = exchange.Features{
+		Supports: exchange.FeaturesSupported{
+			REST:      true,
+			Websocket: true,
+			RESTCapabilities: protocol.Features{
+				TickerFetching:        true,
+				KlineFetching:         true,
+				TradeFetching:         true,
+				OrderbookFetching:     true,
+				AutoPairUpdates:       true,
+				AccountInfo:           true,
+				GetOrder:              true,
+				GetOrders:             true,
+				CancelOrders:          true,
+				CancelOrder:           true,
+				SubmitOrder:           true,
+				UserTradeHistory:      true,
+				CryptoDeposit:         true,
+				CryptoWithdrawal:      true,
+				TradeFee:              true,
+				CryptoWithdrawalFee:   true,
+				MultiChainDeposits:    true,
+				MultiChainWithdrawals: true,
+			},
+			WebsocketCapabilities: protocol.Features{
+				TickerFetching:    true,
+				OrderbookFetching: true,
+			},
+			WithdrawPermissions: exchange.AutoWithdrawCrypto |
+				exchange.AutoWithdrawFiat,
+			Kline: kline.ExchangeCapabilitiesSupported{
+				Intervals: true,
+			},
+		},
+		Enabled: exchange.FeaturesEnabled{
+			AutoPairUpdates: true,
+			Kline: kline.ExchangeCapabilitiesEnabled{
+				Intervals: kline.DeployExchangeIntervals(
```

Comment:
You have not set the ResultLimit and so I receive the following:
`candles count: 1, max limit: 0 request will exceed exchange limits, please reduce start-end time window or use GetHistoricCandlesExtended`